src/component/Calculator.js

```diff
@@ -1,70 +1,65 @@
 /* eslint-disable no-nested-ternary */
-import React from 'react';
+import React, { useState } from 'react';
 import calculate from '../logic/calculate';
 import '../logic/operate';
-import Buttons from './buttonEvents';
+import Events from './buttonEvents';
 
-export default class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      calcObj: {},
-    };
-    this.renderClickable = this.renderClickable.bind(this);
-  }
+const Calculator = () => {
+  const [calcObj, setState] = useState({ total: 0, next: null, operation: null });
 
-  renderClickable(calcObj, text) {
-    this.setState({ calcObj: calculate(calcObj, text) });
-  }
+  const click = (calcObj, text) => {
+    setState({ calcObj: calculate(calcObj, text) });
+  };
 
-  render() {
-    const { calcObj } = this.state;
-    let contentDisplay = '0';
-    if (
-      Object.keys(calcObj).length === 0
-      || (calcObj.total === null
-        && calcObj.next === null
-        && calcObj.operation === null)
-    ) {
-      contentDisplay = '0';
-    } else {
-      contentDisplay = calcObj.next != null ? calcObj.next : calcObj.total;
-    }
+  let contentDisplay = '0';
+  if (
+    Object.keys(calcObj).length === 0
+    || (calcObj.total === null
+      && calcObj.next === null
+      && calcObj.operation === null)
+  ) {
+    contentDisplay = '0';
+  } else {
+    contentDisplay = calcObj.next != null ? calcObj.next
+      : calcObj.next != null ? calcObj.total
+        : calcObj.operation;
+  }
```

Comment:
- [ ] The hook is not working I can't do any calculation.


https://user-images.githubusercontent.com/61562626/140088491-306e67a5-5317-4d40-b313-432fdcfd3d9a.mp4